flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerImplTest.java

```diff
@@ -18,166 +18,218 @@
-		Mockito.when(jobVertex.getJobId()).thenReturn(new JobID());
-		Mockito.when(jobVertex.getJobVertexId()).thenReturn(new JobVertexID());
-		Mockito.when(jobVertex.getGraph()).thenReturn(graph);
-		Mockito.when(jobVertex.getTaskVertices()).thenReturn(taskVertices);
-
-		taskVertices[0] = mockExecutionVertex(jobVertex, 0);
-		taskVertices[1] = mockExecutionVertex(jobVertex, 1);
-		taskVertices[2] = mockExecutionVertex(jobVertex, 2);
-		taskVertices[3] = mockExecutionVertex(jobVertex, 3);
-
-		int numSamples = 100;
-		Time delayBetweenSamples = Time.milliseconds(100L);
-
-		BackPressureStatsTrackerImpl tracker = new BackPressureStatsTrackerImpl(
-				sampleCoordinator, 9999, numSamples, Integer.MAX_VALUE, delayBetweenSamples);
-
-		// getOperatorBackPressureStats triggers stack trace sampling
-		Assert.assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		Mockito.verify(sampleCoordinator, Mockito.times(1)).triggerStackTraceSample(
-				Matchers.eq(taskVertices),
-				Matchers.eq(numSamples),
-				Matchers.eq(delayBetweenSamples),
-				Matchers.eq(BackPressureStatsTrackerImpl.MAX_STACK_TRACE_DEPTH));
-
-		// Request back pressure stats again. This should not trigger another sample request
-		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		Mockito.verify(sampleCoordinator, Mockito.times(1)).triggerStackTraceSample(
-				Matchers.eq(taskVertices),
-				Matchers.eq(numSamples),
-				Matchers.eq(delayBetweenSamples),
-				Matchers.eq(BackPressureStatsTrackerImpl.MAX_STACK_TRACE_DEPTH));
-
-		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		// Complete the future
-		Map<ExecutionAttemptID, List<StackTraceElement[]>> traces = new HashMap<>();
-		for (ExecutionVertex vertex : taskVertices) {
-			List<StackTraceElement[]> taskTraces = new ArrayList<>();
+	public void testGetOperatorBackPressureStats() throws Exception {
+		final ExecutionJobVertex executionJobVertex = BackPressureTrackerTestUtils.createExecutionJobVertex();
+		final ExecutionVertex[] taskVertices = executionJobVertex.getTaskVertices();
 
-			for (int i = 0; i < taskVertices.length; i++) {
-				// Traces until sub task index are back pressured
-				taskTraces.add(createStackTrace(i <= vertex.getParallelSubtaskIndex()));
-			}
+		final int requestId = 0;
+		final long startTime = System.currentTimeMillis();
+		final long endTime = startTime + 1;
+		final double backPressureRatio = 0.1;
 
-			traces.put(vertex.getCurrentExecutionAttempt().getAttemptId(), taskTraces);
-		}
+		final BackPressureStats backPressureStats = createBackPressureStats(
+			taskVertices, requestId, startTime, endTime, backPressureRatio);
+		final BackPressureStatsTracker tracker = createBackPressureTracker(600000, 10000, backPressureStats);
 
-		int sampleId = 1231;
-		int endTime = 841;
+		// trigger back pressure stats request
+		tracker.getOperatorBackPressureStats(executionJobVertex);
```

Comment:
It seems strange to not verify the results of this `getOperatorBackPressureStats`. Actually this check should be done here instead of in `testOperatorBackPressureStatsUpdate` did.